controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,188 @@
+package storagecluster
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	providerClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	// getExternalConfigFromProvider func sets value for this once and use same again
+	externalOCSResources map[types.UID][]ExternalResource
```

Comment:
Empty map is defined, required to initialize 

`externalOCSResources = map[types.UID][]ExternalResource{}`

Reply:
made changes